src/SingleProject/Resizetizer/src/AndroidAdaptiveIconGenerator.cs

```diff
@@ -42,57 +42,110 @@ public IEnumerable<ResizedImageInfo> Generate()
 			var fullIntermediateOutputPath = new DirectoryInfo(IntermediateOutputPath);
 
 			var backgroundFile = Info.Filename;
-			var backgroundIsVector = Info.IsVector;
+			var backgroundIsVector = UseVectors && Info.IsVector;
+			var backgroundExt = backgroundIsVector ? ".xml" : ".png";
+			var backgroundDestFilename = AppIconName + "_background" + backgroundExt;
 
 			var foregroundFile = Info.ForegroundFilename;
-			var foregroundIsVector = Info.ForegroundIsVector;
 			var foregroundExists = File.Exists(foregroundFile);
+			var foregroundIsVector = !foregroundExists || (UseVectors && Info.ForegroundIsVector);
+			var foregroundExt = foregroundIsVector ? ".xml" : ".png";
+			var foregroundDestFilename = AppIconName + "_foreground" + foregroundExt;
 
-			Logger.Log("Looking for Foreground File: " + foregroundFile);
+			if (backgroundIsVector)
+			{
+				Logger.Log("Converting Background SVG to Android Drawable Vector: " + backgroundFile);
 
-			// If we have vectors we can emit an adaptive icon
-			if (backgroundIsVector && (foregroundIsVector || !foregroundExists))
+				var dir = Path.Combine(fullIntermediateOutputPath.FullName, "drawable-v24");
+				var destination = Path.Combine(dir, backgroundDestFilename);
+				if (!Directory.Exists(dir))
+					Directory.CreateDirectory(dir);
```

Comment:
Ah another idea, you can just call `Directory.CreateDirectory(dir);` and if it doesn't exist it's a no-op.
```suggestion
				Directory.CreateDirectory(dir);
```

Reply:
Nice!